there is another error:

```
/Users/runner/work/milvus/milvus/internal/core/src/common/ChunkTarget.cpp:19:12: error: expected unqualified-id
const auto PAGE_SIZE = sysconf(_SC_PAGE_SIZE);
```

--- a/internal/core/src/common/ChunkTarget.cpp
+++ b/internal/core/src/common/ChunkTarget.cpp
@@ -11,12 +11,13 @@
 
 #include <common/ChunkTarget.h>
 #include <algorithm>
+#include <cstdint>
 #include <cstring>
 #include "common/EasyAssert.h"
 #include <sys/mman.h>
 #include <unistd.h>
 
-const auto PAGE_SIZE = sysconf(_SC_PAGE_SIZE);
+const uint32_t PAGE_SIZE = sysconf(_SC_PAGE_SIZE);
 namespace milvus {
 void
 MemChunkTarget::write(const void* data, size_t size, bool append) {

--- a/internal/core/src/common/ChunkTarget.cpp
+++ b/internal/core/src/common/ChunkTarget.cpp
@@ -17,7 +17,7 @@
 #include <sys/mman.h>
 #include <unistd.h>
 
-const uint32_t PAGE_SIZE = sysconf(_SC_PAGE_SIZE);
+const uint32_t SYS_PAGE_SIZE = sysconf(_SC_PAGE_SIZE);
 namespace milvus {
 void
 MemChunkTarget::write(const void* data, size_t size, bool append) {
@@ -95,7 +95,8 @@ std::pair<char*, size_t>
 MmapChunkTarget::get() {
     // Write padding to align with the page size, ensuring the offset_ aligns with the page size.
     auto padding_size =
-        (size_ / PAGE_SIZE + (size_ % PAGE_SIZE != 0)) * PAGE_SIZE - size_;
+        (size_ / SYS_PAGE_SIZE + (size_ % SYS_PAGE_SIZE != 0)) * SYS_PAGE_SIZE -
+        size_;
     char padding[padding_size];
     memset(padding, 0, sizeof(padding));
     write(padding, padding_size);

--- a/internal/core/src/mmap/ChunkedColumn.h
+++ b/internal/core/src/mmap/ChunkedColumn.h
@@ -353,8 +353,8 @@ class ChunkedVariableColumn : public ChunkedColumnBase {
                 {chunks_[start_chunk_id]->Data(),
                  std::dynamic_pointer_cast<StringChunk>(chunks_[start_chunk_id])
                      ->Offsets(),
-                 start_offset_in_chunk,
-                 start_offset_in_chunk + len});
+                 static_cast<int>(start_offset_in_chunk),
+                 static_cast<int>(start_offset_in_chunk + len)});
 
             start_offset_in_chunk = 0;
             length -= len;

